### PR TITLE
change get param from bool to int

### PR DESCRIPTION
**Pull request by totto82**
An embarrassing bug in #3355. The testing was done on #3331, where the limit was hard-coded to 3.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by bska** [comment]
Does this affect the convergence of the `ACTIONX_M1` model at all?

**Comment by totto82** [comment]
> Does this affect the convergence of the ACTIONX_M1 model at all?

With bool the parameter was hard-coded to 1. For MSW that means going from all to 1 and for STW it means going from none to 1. I will have to go through the test cases again. In particular the STW. The MSW tests we now know both works for 1 and all, so it would be very surprising if it does not work for 3. The results should not change, but I need to make sure that all run smoothly.

**Comment by totto82** [comment]
```
    mpi.compareECLFiles_flow+SPE1CASE2_2P ok
    mpi.compareECLFiles_flow+SPE1CASE1 ok
    mpi.compareECLFiles_flow+SPE1CASE1_IMPORT (124 != 125)
    mpi.compareECLFiles_flow+SPE1CASE2_THERMAL ok
    mpi.compareECLFiles_flow+SPE1CASE2_ROCK2DTR (153 != 143)
    mpi.compareECLFiles_flow+SPE1CASE2_THERMAL_WATVISC ok
    mpi.compareECLFiles_flow+SPE1CASE1_BRINE ok
    mpi.compareECLFiles_flow+SPE1CASE1_METRIC_VFP1 ok 
    mpi.compareECLFiles_flow+SPIDER_CAKESLICE (36 != 33)
    mpi.compareECLFiles_flow+SPE9_CP_SHORT ok
    mpi.compareECLFiles_flow+2D_OILWATER_POLYMER ok
    mpi.compareECLFiles_flow+SPE9_CP_GROUP_RESV ok
    mpi.compareECLFiles_flow+SPE9_CP_GROUP ok
    mpi.compareECLFiles_flow+3D_MSW ok
    mpi.compareECLFiles_flow+2D_POLYMER_INJECTIVITY (120 != 113)
    mpi.compareECLFiles_flow+2D_THREEPHASE_POLY_HETER (253 != 252)
    mpi.compareECLFiles_flow+SPE5CASE1 ok
    mpi.compareECLFiles_flow+SPE5CASE1_DYN ok
    mpi.compareECLFiles_flow+BASE_MODEL_1 ok
    mpi.compareECLFiles_flow+MSW_MODEL_1 ok
    mpi.compareECLFiles_flow+3D_WECON ok
    mpi.compareECLFiles_flow+FAULTS_MODEL_1 ok
    mpi.compareECLFiles_flow+0A1_GRCTRL_LRAT_ORAT_BASE_MODEL2_STW (11 != 10)
    mpi.compareECLFiles_flow+0A2_GRCTRL_LRAT_ORAT_GGR_BASE_MODEL2_STW ok
    mpi.compareECLFiles_flow+0A1_GRCTRL_LRAT_ORAT_BASE_MODEL2_MSW ok
    mpi.compareECLFiles_flow+0A2_GRCTRL_LRAT_ORAT_GGR_BASE_MODEL2_MSW ok
    mpi.compareECLFiles_flow+0A3_GRCTRL_LRAT_LRAT_BASE_MODEL2_STW ok
    mpi.compareECLFiles_flow+0A3_GRCTRL_LRAT_LRAT_BASE_MODEL2_MSW ok
    mpi.compareECLFiles_flow+0A4_GRCTRL_LRAT_LRAT_GGR_BASE_MODEL2_STW (9 != 10)
    mpi.compareECLFiles_flow+UDQ_ACTIONX (32 != 33)
    mpi.compareECLFiles_flow+0A4_GRCTRL_LRAT_LRAT_GGR_BASE_MODEL2_MSW ok
    mpi.compareECLFiles_flow+UDQ_WCONPROD ok
    mpi.compareECLFiles_flow+0_BASE_MODEL2 (67 != 63)
    mpi.compareECLFiles_flow+ACTIONX_M1 ok
    mpi.compareECLFiles_flow+UDQ_M2 (20 != 19)
    mpi.compareECLFiles_flow+UDQ_M3 (24 != 26) 
    mpi.compareECLFiles_flow+UDQ_M1 (52 != 48)
    mpi.compareECLFiles_flow+PYACTION_WCONPROD ok
    mpi.compareECLFiles_flow+1_MULTREGT_MODEL2 (70 != 71)
    mpi.compareECLFiles_flow+2_MULTXYZ_MODEL2 (59 != 58)
    mpi.compareECLFiles_flow+3_MULTFLT_MODEL2 ok
    mpi.compareECLFiles_flow+4_MINPVV_MODEL2 ok
    mpi.compareECLFiles_flow+5_SWATINIT_MODEL2 (64 != 58)
    mpi.compareECLFiles_flow+6_ENDSCALE_MODEL2 (64 != 59)
    mpi.compareECLFiles_flow+7_HYSTERESIS_MODEL2 (65 != 63)
    mpi.compareECLFiles_flow+9_EDITNNC_MODEL2 (62 != 61)
    mpi.compareECLFiles_flow+8_MULTIPLY_TRANXYZ_MODEL2 (64 != 68)
    mpi.compareECLFiles_flow+9_1A_DEPL_MAX_RATE_MIN_BHP_STW ok
    mpi.compareECLFiles_flow+9_1A_DEPL_MAX_RATE_MIN_BHP_MSW (22 != 23)
    mpi.compareECLFiles_flow+9_1B_DEPL_MAX_RATE_MIN_THP_STW (32 != 31)
    mpi.compareECLFiles_flow+9_2A_DEPL_GCONPROD_1L_STW (31 != 32)
    mpi.compareECLFiles_flow+9_2B_DEPL_GCONPROD_2L_STW ok
    mpi.compareECLFiles_flow+9_2A_DEPL_GCONPROD_1L_MSW ok
    mpi.compareECLFiles_flow+9_2B_DEPL_GCONPROD_2L_MSW ok
    mpi.compareECLFiles_flow+9_3A_GINJ_REIN-G_STW (32 != 34)
    mpi.compareECLFiles_flow+9_1B_DEPL_MAX_RATE_MIN_THP_MSW (31 != 32)
    mpi.compareECLFiles_flow+9_3A_GINJ_REIN-G_MSW (33 != 31)
    mpi.compareECLFiles_flow+9_3B_GINJ_GAS_EXPORT_STW (32 != 34)
    mpi.compareECLFiles_flow+9_3C_GINJ_GAS_GCONSUMP_STW (32 != 31)
    mpi.compareECLFiles_flow+9_3B_GINJ_GAS_EXPORT_MSW (32 != 34)
    mpi.compareECLFiles_flow+9_3D_GINJ_GAS_MAX_EXPORT_STW (36 != 37)
    mpi.compareECLFiles_flow+9_3C_GINJ_GAS_GCONSUMP_MSW (33 != 32)
    mpi.compareECLFiles_flow+9_3E_GAS_MIN_EXPORT_STW ok
    mpi.compareECLFiles_flow+9_3D_GINJ_GAS_MAX_EXPORT_MSW (32 != 33)
    mpi.compareECLFiles_flow+9_3E_GAS_MIN_EXPORT_MSW ok
    mpi.compareECLFiles_flow+9_4A_WINJ_MAXWRATES_MAXBHP_GCONPROD_1L_STW (30 != 31)
    mpi.compareECLFiles_flow+9_4B_WINJ_VREP-W_STW (31 != 34)
    mpi.compareECLFiles_flow+9_4A_WINJ_MAXWRATES_MAXBHP_GCONPROD_1L_MSW (32 != 31)
    mpi.compareECLFiles_flow+9_4B_WINJ_VREP-W_MSW (34 != 32)
    mpi.compareECLFiles_flow+9_4C_WINJ_GINJ_VREP-W_REIN-G_STW (45 != 38)
    mpi.compareECLFiles_flow+9_4D_WINJ_GINJ_GAS_EXPORT_STW (45 != 38)
    mpi.compareECLFiles_flow+9_4C_WINJ_GINJ_VREP-W_REIN-G_MSW (32 != 33)
    mpi.compareECLFiles_flow+1_MSW_MODEL6 (37 != 32)
    mpi.compareECLFiles_flow+9_4D_WINJ_GINJ_GAS_EXPORT_MSW ok
    mpi.compareECLFiles_flow+BASE_MSW_WSEGAICD ok
    mpi.compareECLFiles_flow+NNC_AND_EDITNNC ok
    mpi.compareECLFiles_flow+TEST_WSEGSICD ok
    mpi.compareECLFiles_flow+SPE1FOAM (124 != 123)
    mpi.compareECLFiles_flow+COMPL_SMRY ok
    mpi.compareECLFiles_flow+3D_TRAN_OPERATOR ok
    mpi.compareECLFiles_flow+MOD4_UDQ_ACTIONX (98 != 94)
    mpi.compareECLFiles_flow+MOD4_GRP (97 != 107)
```

**Comment by totto82** [comment]
The test look ok. Some hard cases get significant better in terms of convergence. Most cases is not much affected. The only case with significant more iterations is MOD4_GRP.  The current default of 3 may not be optimal and we will continue to test this parameter. It is a compromise between always and never solving the inner iterations needed to get ok performance for model2. If we in the future are able to nail the issue that makes inner iterations problematic for model2 we should change the default to 99 or remove the options as also @GitPaean pointed out in the earlier discussions.

**Comment by bska** [comment]
> The MSW tests we now know both works for 1 and all, so it would be very surprising if it does not work for 3

Sure.  I was more alluding to the fact that we switched to using [non-strict tolerances](https://github.com/OPM/opm-simulators/pull/3350/commits/a55fa81b2fba3d47c15a0e64eacc666314bc6adf) for the parallel `ACTIONX_M1` test in #3350 and I just wondered whether this change would be enough to restore the default tolerances.

**Comment by totto82** [comment]
Can I start an update for this. The current state of the simulator with hard-coded 1 for this parameter is very problematic.

**Comment by totto82** [comment]
> Sure. I was more alluding to the fact that we switched to using non-strict tolerances for the parallel ACTIONX_M1 test in #3350 and I just wondered whether this change would be enough to restore the default tolerances.

I noticed that most of the parallel tests run with the non-strict tolerances. Maybe we are able to restore the non-strict tolerance for ACTION_M1 for this PR, but wouldn't it just be a matter of time before we have to relax it again like the other parallel tests?

**Review by bska — APPROVED** [review]
Looks good.  Sorry that I didn't catch that whilst reviewing #3350.

**Comment by bska** [comment]
> Maybe we are able to restore the non-strict tolerance for ACTION_M1 for this PR, but wouldn't it just be a matter of time before we have to relax it again like the other parallel tests?

I suppose so.  Okay, I'll update the reference solutions and merge this.

**Comment by bska** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=553 please

**Comment by joakim-hove** [comment]
OK - this bites me at the moment; it is approved and green. I'll merge - even if I don't really understand this.

**Comment by bska** [comment]
> OK - this bites me at the moment; it is approved and green. I'll merge - even if I don't really understand this.

For the record: I was waiting for the updated reference solutions to get installed on the CI system.  Merging before that is likely to create a race condition and making the post-merge build fail.

**Comment by joakim-hove** [comment]
> For the record: I was waiting for the updated reference solutions to get installed on the CI system. Merging before that is likely to create a race condition and making the post-merge build fail.

OK - I have never really understood these ordering requirements; I have always assumed that the build server clones opm-tests and then the race-window should at maximum be of order seconds. Or do I misunderstand here?

**Comment by akva2** [comment]
no, it does not work like that. opm-tests is such a big repo and since there are limits to the amount of network traffic on the jenkins servers, all builds use a shared copy from the job opm-tests. if this has not run before the post-builder for opm-simulators kicks off, it will use outdated data. the opm-tests job polls at 5 min intervals, the post-builders at 15 min intervals to try to reduce the amount of issues. but there will always be issues due to having to use the shared copy.

**Comment by joakim-hove** [comment]
> the opm-tests job polls at 5 min intervals

OK - I understand. So the correct drill is to merge the opm-tests PR; wait for 5+ minutes and then merge the other PR's?

**Comment by bska** [comment]
> So the correct drill is to merge the opm-tests PR; wait for 5+ minutes and then merge the other PR's?

Almost.  The correct drill is to wait until the `opm-tests` post-merge build has actually run on the CI system before merging the others.

**Comment by joakim-hove** [comment]
> The correct drill is to wait until the opm-tests post-merge build has actually run on the CI system before merging the others.

OK - and the opm-tests post-merge build is that the job actually installing the updated reference data on the CI system? Maybe we could get a push notification in opm-tests repo when that job has been completed?

**Comment by bska** [comment]
> > he correct drill is to wait until the opm-tests post-merge build has actually run on the CI system before merging the others.
> 
> OK - and the opm-tests post-merge build is that the job actually installing the updated reference data on the CI system?

Yes it is.

> Maybe we could get a push notification in opm-tests repo when that job has been completed?

Maybe.  I don't know what's possible/convenient in the GitHub API.  Would you like to comment, @akva2?

**Comment by akva2** [comment]
i am sure we could probably set up push notification support for opm-tests, but i do not know exactly what it would entail at this point.

**Comment by akva2** [comment]
tell me before you merge then next opm-tests update and we can experiment a little. does not look to difficult, but well, it always looks simple up front ...

**Comment by bska** [comment]
> tell me before you merge then next opm-tests update and we can experiment a little.

I will.  I was about to review OPM/opm-tests#552 now anyway.

**Pull request merged**
